src/shared/generator.c

```diff
@@ -166,6 +166,15 @@ int generator_write_fsck_deps(
                 return 0;
         }
 
+        r = fsck_available();
+        if (r < 0)
+                log_warning_errno(r, "Checking was requested for %s, but couldn't detect if the fsck command may be used, proceeding: %m", what);
+        else if (r == 0) {
+                /* treat missing fsck as essentially OK */
+                log_debug("Checking was requested for %s, but the fsck command does not exist.", what);
+                return 0;
+        }
```

Comment:
and once the changes proposed above are made, please move this so that it is only executed if fstype is not known.